internal/controller/helmrepository_controller_test.go

```diff
@@ -642,28 +643,29 @@ func TestHelmRepositoryReconciler_reconcileSource(t *testing.T) {
 			want: sreconcile.ResultSuccess,
 		},
 		{
-			name:     "Stored index with different digest and same revision",
+			name:     "Stored index with different revision and same digest",
```

Comment:
> But I think the test case should remain the same with the same expected conditions.

Previously, the test was checking that changing the digest and keeping the revision the same WILL still result in Loading the index. The `reconcileSource` function has changed to only check if the revision has changed(doesn't check the digest at all since at this point we haven't loaded the index) so this is no longer the case. If only you change the digest but keep the revision the same, it will no longer result in a new index. The check for digest now only happens in `ReconcileArtifact`.

Reply:
> Same or a different digest shouldn't matter anymore here, right?

Yes, I will take out the test case, it is no longer needed